ext/vulnsrc/redhat/redhat.go

```diff
@@ -0,0 +1,684 @@
+	var vulnIDs []database.VulnerabilityID
+	rhsaName := ParseRhsaName(advisoryDefinition)
+	cveNames := ParseCveNames(advisoryDefinition)
+	for _, cveName := range cveNames {
+		vulnID := database.VulnerabilityID{Name: cveName + " - " + rhsaName, Namespace: ParseVulnerabilityNamespace(advisoryDefinition)}
+		vulnIDs = append(vulnIDs, vulnID)
+	}
+	return vulnIDs
+}
+
+// parse the CVE name(s) (e.g.: "CVE-2019-11249") from the given advisory definition
+func ParseCveNames(advisoryDefinition ParsedAdvisory) []string {
+	var cveNames []string
+	for _, cve := range advisoryDefinition.Metadata.Advisory.CveList {
+		cveNames = append(cveNames, cve.Value)
+	}
+	return cveNames
+}
+
+// parse the RHSA name (e.g.: "RHBA-2019:2794") from the given advisory definition
+func ParseRhsaName(advisoryDefinition ParsedAdvisory) string {
+	return strings.TrimSpace(advisoryDefinition.Metadata.Title[:strings.Index(advisoryDefinition.Metadata.Title, ": ")])
+}
+
+// parse the namespace from the given advisory definition
+func ParseVulnerabilityNamespace(advisoryDefinition ParsedAdvisory) string {
+	// use criteria parse result
+	moduleNamespaces := ParseCriteriaForModuleNamespaces(advisoryDefinition.Criteria)
+	if len(moduleNamespaces) > 0 {
+		// use MODULE namespace
+		return moduleNamespaces[0]
+	} else {
+		// use CPE namespace
+		cpeNames, err := ParseCpeNamesFromAffectedCpeList(advisoryDefinition.Metadata.Advisory.AffectedCpeList)
+		if err != nil {
+			// log error and continue
+			log.Error(err)
+			return ""
+		}
+		if len(cpeNames) == 0 {
+			// no namespace found
+			return ""
+		} else {
+			return cpeNames[0]
+		}
+	}
+}
+
+func GetSeverity(severity string) database.Severity {
+	switch strings.Title(strings.ToLower(severity)) {
+	case "None":
+		return database.NegligibleSeverity
+	case "Low":
+		return database.LowSeverity
+	case "Moderate":
+		return database.MediumSeverity
+	case "Important":
+		return database.HighSeverity
+	case "Critical":
+		return database.CriticalSeverity
+	case "Unknown":
+		return database.UnknownSeverity
+	default:
+		log.Warningf("could not determine vulnerability severity from: %s.", severity)
+		return database.UnknownSeverity
+	}
+}
+
+// checks whether the given severity is significant (used to determine whether vulns will be parsed and stored for it)
+func IsSignificantSeverity(severity string) bool {
+	switch strings.Title(strings.ToLower(severity)) {
+		case "None":
+			return false
+	default:
+		// anything else is considered significant
+		return true
+	}
+}
+
+func extractAllCriterions(criteria OvalV2Criteria) []OvalV2Criterion {
+    var criterions []OvalV2Criterion
+    for _, criterion := range criteria.Criteria {
+        // recursively append criteria contents
+        criterions = append(criterions, extractAllCriterions(criterion)...)
+    }
+    for _, criterion := range criteria.Criterion {
+        // append criterion
+        criterions = append(criterions, criterion)
```

Comment:
It probably makes sense to append criterion only
`if strings.Contains(criterion.Comment, "is earlier than") `